src/persist-client/src/impl/machine.rs

```diff
@@ -297,7 +296,11 @@ where
                     );
                     self.update_state(current).await;
 
-                    // TODO: Some sort of exponential backoff here?
+                    // TODO: Some sort of exponential backoff here? We have
```

Comment:
Hmm? :thinking: Hanging tests seems ... not good.

Reply:
I cargo stressed for a while and nothing seems to hang unless I add the retry here. I suspect something I don't understand about async or tokio, but given that it sorta doesn't make sense to add it here anyway, I don't feel the need to immediately dig into it /shrug

Reply:
Aha! Tried the same diff on top of #12301 which fixes the futures_executor::block_on and that worked. Makes sense since that PR fixes a deadlock